docs/migration/cosmoshub-4-v9-Lambda-upgrade.md

```diff
@@ -0,0 +1,297 @@
+                - [Expected upgrade result](#expected-upgrade-result)
+            - [Auto-Downloading the Gaia v9.0.0 binary (not recommended!)](#auto-downloading-the-gaia-v900-binary-not-recommended)
+                - [Preparation](#preparation-1)
+                - [Expected result](#expected-result)
+    - [Upgrade duration](#upgrade-duration)
+    - [Rollback plan](#rollback-plan)
+    - [Communications](#communications)
+    - [Risks](#risks)
+    - [Reference](#reference)
+
+## On-chain governance proposal attains consensus
+
+[Proposal #187](https://www.mintscan.io/cosmos/proposals/187) is the reference on-chain governance proposal for this upgrade, which is still in its voting period. Neither core developers nor core funding entities control the governance, and this governance proposal has passed in a _fully decentralized_ way.
+
+## Upgrade will take place March 15, 2023
+
+The upgrade will take place at a block height of `14470501`. At the time of writing, and at current block times (around 7s/block), this block height corresponds approximately to `Wednesday, 15-March-23 12:52:12 CET`. This date/time is approximate as blocks are not generated at a constant interval. You can stay up-to-date using this [live countdown](https://www.mintscan.io/cosmos/blocks/14470501) page.
+
+## Chain-id will remain the same
+
+The chain-id of the network will remain the same, `cosmoshub-4`. This is because an in-place migration of state will take place, i.e., this upgrade does not export any state.
+
+## Preparing for the upgrade
+
+### System requirement
+
+32GB RAM is recommended to ensure a smooth upgrade.
+
+If you have less than 32GB RAM, you might try creating a swapfile to swap an idle program onto the hard disk to free up memory. This can
+allow your machine to run the binary than it could run in RAM alone.
+
+```shell
+sudo fallocate -l 16G /swapfile
+sudo chmod 600 /swapfile
+sudo mkswap /swapfile
+sudo swapon /swapfile
+```
+
+### Backups
+
+Prior to the upgrade, validators are encouraged to take a full data snapshot. Snapshotting depends heavily on infrastructure, but generally this can be done by backing up the `.gaia` directory.
+If you use Cosmovisor to upgrade, by default, Cosmovisor will backup your data upon upgrade. See below [upgrade by cosmovisor](#method-ii-upgrade-using-cosmovisor-by-manually-preparing-the-gaia-v700-binary) section.
+
+It is critically important for validator operators to back-up the `.gaia/data/priv_validator_state.json` file after stopping the gaiad process. This file is updated every block as your validator participates in consensus rounds. It is a critical file needed to prevent double-signing, in case the upgrade fails and the previous chain needs to be restarted.
+
+### Testing
+
+For those validator and full node operators that are interested in ensuring preparedness for the impending upgrade, you can run a [v8-Rho local testnet](https://github.com/cosmos/testnets/tree/master/local).
```

Comment:
```suggestion
For those validator and full node operators that are interested in ensuring preparedness for the impending upgrade, you can run a [v8-Rho local testnet](https://github.com/cosmos/testnets/tree/master/local) or join in our [v9-Rho public-testnet](https://github.com/cosmos/testnets/tree/master/public).
```

Reply:
@dasanchez could we have a https://github.com/cosmos/testnets/tree/master/local/previous-local-testnets/rho in place?

Reply:
@mmulji-ic here you go: https://github.com/cosmos/testnets/tree/master/local/previous-local-testnets/rho